api/src/convert/signed_contingent_input.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Convert to/from external::SignedContingentInput.
+
+use crate::{external, ConversionError};
+use mc_transaction_core::{ring_signature::RingMLSAG, tx::TxIn};
+use mc_transaction_extra::{SignedContingentInput, UnmaskedAmount};
+
+/// Convert mc_transaction_extra::SignedContingentInput -->
+/// external::SignedContingentInput.
+impl From<&SignedContingentInput> for external::SignedContingentInput {
+    fn from(src: &SignedContingentInput) -> Self {
+        let mut sci = external::SignedContingentInput::new();
+        sci.set_block_version(src.block_version);
+        sci.set_tx_in(external::TxIn::from(&src.tx_in));
+        sci.set_mlsag(external::RingMLSAG::from(&src.mlsag));
+        sci.set_pseudo_output_amount(external::UnmaskedAmount::from(&src.pseudo_output_amount));
+        sci.set_required_output_amounts(
+            src.required_output_amounts
+                .iter()
+                .map(external::UnmaskedAmount::from)
+                .collect(),
+        );
+        sci.set_tx_out_global_indices(src.tx_out_global_indices.clone());
+
+        sci
+    }
+}
+
+/// Convert external::SignedContingentInput -->
+/// mc_transaction_extra::SignedContingentInput.
+impl TryFrom<&external::SignedContingentInput> for SignedContingentInput {
+    type Error = ConversionError;
+
+    fn try_from(src: &external::SignedContingentInput) -> Result<Self, Self::Error> {
+        let block_version = src.get_block_version();
+        let tx_in = TxIn::try_from(src.get_tx_in())?;
+        let mlsag = RingMLSAG::try_from(src.get_mlsag())?;
+        let pseudo_output_amount = UnmaskedAmount::try_from(src.get_pseudo_output_amount())?;
+        let required_output_amounts = src
+            .get_required_output_amounts()
+            .iter()
+            .map(UnmaskedAmount::try_from)
+            .collect::<Result<Vec<UnmaskedAmount>, ConversionError>>()?;
```

Comment:
:upside_down_face: : I have no objection to leaving this explicit.  I generally omit (probably  me being lazy)
```suggestion
            .collect::<Result<Vec<_>, _>>()?;
```